indexer/Indexer.cc

```diff
@@ -356,6 +357,76 @@ void TuIndexer::saveNamespaceDecl(const clang::NamespaceDecl *namespaceDecl) {
   doc.symbolInfos.insert({symbol, scip::SymbolInformation{}});
 }
 
+void TuIndexer::saveNestedNameSpecifier(
+    const clang::NestedNameSpecifierLoc &argNameSpecLoc) {
+  clang::NestedNameSpecifierLoc nameSpecLoc = argNameSpecLoc;
+  while (nameSpecLoc.hasQualifier()) {
+    auto nameSpec = nameSpecLoc.getNestedNameSpecifier();
+    using Kind = clang::NestedNameSpecifier;
+    switch (nameSpec->getKind()) {
+    case Kind::Namespace: {
+      auto namespaceDecl = nameSpec->getAsNamespace();
+      std::string_view symbol;
+      if (auto optSymbol =
+              this->symbolFormatter.getNamespaceSymbol(namespaceDecl)) {
+        symbol = optSymbol.value();
+      } else {
+        break;
+      }
+      scip::Occurrence occ;
+      auto [range, fileId] = FileLocalSourceRange::fromNonEmpty(
+          this->sourceManager, nameSpecLoc.getLocalSourceRange());
+      range.addToOccurrence(occ);
+      occ.set_symbol(symbol.data(), symbol.size());
+      auto &doc = this->documentMap[{fileId}];
+      doc.occurrences.emplace_back(scip::OccurrenceExt{std::move(occ)});
+      break;
+    }
+    // TODO: Handle all these other cases too
```

Comment:
Maybe we should just handle all of these cases right now instead of having a TODO